### PR TITLE
WebSockets refactoring with callbacks

### PR DESCRIPTION
**Pull request by acatinon**
Still a WIP for now, but feedback is welcome :)

### CODE REVIEW

**Comment by jankjr** [comment]
oh I really like this interface :)

**Comment by acatinon** [comment]
Finally I think this is ready to be reviewed

File: src/nash/mod.rs [hunk 1]
```diff
@@ -706,73 +711,104 @@ impl Stream for NashStream {
...
 
-impl From<nash_protocol::protocol::subscriptions::SubscriptionResponse>
-    for OpenLimitsWebsocketMessage
-{
-    fn from(message: nash_protocol::protocol::subscriptions::SubscriptionResponse) -> Self {
-        match message {
-            nash_protocol::protocol::subscriptions::SubscriptionResponse::Orderbook(resp) => {
-                OpenLimitsWebsocketMessage::OrderBook(OrderBookResponse {
+#[derive(Debug)]
+pub struct SubscriptionResponseWrapper(SubscriptionResponse);
```
Comment (acatinon) [inline src/nash/mod.rs:780]: It would be nice to make `nash_protocol::protocol::SubscriptionResponse` impl `Clone` to avoid this wrapper

Comment (Ejhfast) [reply src/nash/mod.rs:780]: yes, can definitely do that!

**Comment by Ejhfast** [comment]
Awesome @acatinon, I've been taking a look this evening. It looks great so far, though I'm having some trouble verifying that the Nash subscriptions are working under the callback method. Could just be doing something dumb, will keep playing with it.

**Comment by acatinon** [comment]
@Ejhfast I'm sure you have more important things to do, I will have a look tonight

**Comment by acatinon** [comment]
@Ejhfast The Nash WS tests should be ok

**Comment by Ejhfast** [comment]
Yes, looks good!  🎉 I was doing something dumb, typo in my code. Thanks for updating the tests, all good on my end to merge this. 

Can do a later small PR to fix `SubscriptionResponseWrapper` once the nash client is updated.

**Review by Ejhfast — APPROVED** [review]

**Pull request merged**